frontend/apps/app/components/PublicSessionDetailPage/PublicSessionDetailPage.tsx

```diff
@@ -0,0 +1,150 @@
+
+  // First check if the session is publicly shared
+  const { data: publicShareData } = await supabase
+    .from('public_share_settings')
+    .select('design_session_id')
+    .eq('design_session_id', designSessionId)
+    .single()
+
+  if (!publicShareData) {
+    notFound()
+  }
+
+  // Fetch design session data directly from tables
+  const { data: designSession, error: sessionError } = await supabase
+    .from('design_sessions')
+    .select('id, name, created_at, parent_design_session_id')
+    .eq('id', designSessionId)
+    .single()
+
+  if (sessionError || !designSession) {
+    notFound()
+  }
+
+  // Fetch timeline items
+  const { data: timelineItems } = await supabase
+    .from('timeline_items')
+    .select(
+      'id, design_session_id, content, created_at, updated_at, building_schema_version_id, type, query_result_id, assistant_role',
+    )
+    .eq('design_session_id', designSessionId)
+    .order('created_at', { ascending: true })
+
+  // Fetch building schema
+  const { data: buildingSchemas } = await supabase
+    .from('building_schemas')
+    .select(
+      'id, design_session_id, schema, created_at, git_sha, initial_schema_snapshot, schema_file_path',
+    )
+    .eq('design_session_id', designSessionId)
+
+  if (!buildingSchemas || buildingSchemas.length === 0) {
+    notFound()
+  }
+
+  const buildingSchema = buildingSchemas[0]
+  if (!buildingSchema || !buildingSchema.id) {
+    notFound()
+  }
+  const buildingSchemaId = buildingSchema.id
+
+  // Fetch versions
+  const { data: versions } = await supabase
+    .from('building_schema_versions')
+    .select('id, building_schema_id, number, created_at, patch, reverse_patch')
+    .eq('building_schema_id', buildingSchemaId)
+    .order('number', { ascending: false })
+
+  if (!versions || versions.length === 0) {
+    notFound()
+  }
+
+  // Parse schema
+  const parsedSchema = safeParse(schemaSchema, buildingSchema.schema)
+  if (!parsedSchema.success) {
+    notFound()
+  }
+  const initialSchema = parsedSchema.output
+
+  // Get latest version for prev schema
+  const latestVersion = versions[0]
+  if (!latestVersion || !latestVersion.id || latestVersion.number === null) {
+    notFound()
+  }
+
+  const initialPrevSchema =
+    (await buildPrevSchema({
+      currentSchema: initialSchema,
+      currentVersionId: latestVersion.id,
+    })) ?? initialSchema
+
+  // Add required fields for timeline items and design session
+  const designSessionWithTimelineItems = {
+    id: designSession.id ?? '',
+    organization_id: 'public', // Dummy value for public access
+    timeline_items: (timelineItems || []).map((item) => ({
+      ...item,
+      id: item.id ?? '',
+      content: item.content ?? '',
+      type: item.type ?? 'user',
+      created_at: item.created_at ?? new Date().toISOString(),
+      design_session_id: item.design_session_id ?? designSessionId,
+      organization_id: 'public', // Dummy value for public access
```

Comment:
I see 👀 If this organization_id is not used by SessionDetailPageClient, I would like to remove it from the type, but since timeline_items will be removed and replaced by LangGraph messages in the future, it can wait for now.